src/main/java/seedu/bookmark/model/book/Note.java

```diff
@@ -0,0 +1,56 @@
+package seedu.bookmark.model.book;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.bookmark.commons.util.AppUtil.checkArgument;
+
+public class Note {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Note title and text should only contain alphanumeric characters and spaces, and it should not be blank";
+    /*
+     * The first character of the note must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
```

Comment:
Should we allow more than alphanumeric characters? I think users would want to use punctuation when writing down their thoughts on the book.

Reply:
Also I think it would be good to put in some length limit for the title and text as well. I don't think the application would be able to handle displaying text that is for example, 10,000 words long well. Perhaps something like 120 characters for the title and 1000 characters for the text?

Reply:
I was thinking of that too! I have implemented a regex to handle these constraints. Symbols are now supported, and title has to be within 1 to 120 characters, text has to be within 1 to 1000 characters.